### PR TITLE
fix acl query template

### PR DESCRIPTION
**Pull request by seanstory**
## Closes https://github.com/elastic/connectors/issues/1696

In a review of the [DLS + Internal Knowledge Search blog](https://www.elastic.co/search-labs/blog/articles/dls-internal-knowledge-search), @xeraa [had commented](https://github.com/elastic/search-labs-elastic-co/pull/292#discussion_r1467288194) that our DLS query looked overly complex. This sent me down a rabbit hole, first figuring out that our example DLS app _was_ using an overly-complex query, then wondering why we were having to create an explicit query at all, then finally realizing I'd had these thoughts before and filed https://github.com/elastic/connectors/issues/1696 to eventually address the issue.

The time has come. 

This PR removes the outer `bool: { filter: {` clause that's no longer necessary (since Workplace Search) as we do not have a `_deny_access_control` pattern for any of our connectors (yet). 

It also fixes the syntax of the output document so that the template contents can be directly taken and shoved into a role descriptor.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes

#### Changes Requiring Extra Attention

- [x] Security-related changes

## Related Pull Requests

* related update to the example app: https://github.com/elastic/elasticsearch-labs/pull/18 (we could replace most of the role descriptor once this PR is merged)

## Release Note

Changes the output structure of the documents created by ACL Syncs. Now these docs are appropriately structured to create Role Descriptors. The structure to access the identity and its permissions have not changed. This structure can take up to half as much space to store as the previous structure.

### CODE REVIEW

File: tests/sources/test_google_drive.py [hunk 1]
```diff
@@ -1474,11 +1455,24 @@ async def test_prepare_access_control_doc(user, groups, access_control_doc):
...
             with mock.patch.object(ServiceAccountManager, "refresh"):
-                assert (
-                    access_control_doc
-                    == await source.prepare_single_access_control_document(user=user)
+                result = await source.prepare_single_access_control_document(user=user)
+
+                # mustache templates are hard to verify, so let's strip the mustache
+                filtered_input_source = " ".join(
+                    access_control_doc["query"]["template"]["source"]
+                    .replace("{{toJson}}access_control{{/toJson}}", "[]")
```
Comment (navarone-feekery) [inline tests/sources/test_google_drive.py:1463]: Does the `#` in the toJson template string need to be included in this replace as well?

Something like `.replace("{{#toJson}}access_control{{/toJson}}", "[]")`

File: connectors/access_control.py [hunk 2]
```diff
@@ -5,6 +5,26 @@
...
+                                    "must_not": {
+                                        "exists": {
+                                            "field": "_allow_access_control"
+                                        }
+                                    }
+                                }
+                            },
+                            {
+                                "terms": {
+                                    "_allow_access_control.enum": {{#toJson}}access_control{{/toJson}}
```
Comment (navarone-feekery) [inline connectors/access_control.py:22]: TIL about the [toJson search template](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template.html#search-template-convert-json). Nice implementation!

**Review by navarone-feekery — APPROVED** [review]
LGTM, just one question

File: tests/sources/test_google_drive.py (hunk 1 above)
Comment (seanstory) [reply tests/sources/test_google_drive.py]: Huh. Yes, the `#` should matter. This must not be doing what I thought it was.

Turns out I can remove all the complexity I added to these tests. 🙈

**Comment by seanstory** [comment]
@navarone-feekery can I get a re-approval?

**Review by navarone-feekery — APPROVED** [review]

**Pull request merged**